### DISCUSSION TITLE
Does trpc have n+1 problem like graphql?

### DISCUSSION THREAD

**Discussion by MejanH** [trpc/trpc#3966] [Q&A] [answered]
I really love trpc. solves almost every issue I had with graphql. But just wanted to know since i can't find anything on docs. Should i be careful for the n+1 problem with trpc too?

**Comment by c-ehrlich** [comment, marked as answer]
No. The syntax of tRPC looks like GraphQL but the mental model is closer to REST. You can still encounter N+1 in database queries just as you could anywhere else that you query a database, but it's not a fundamental consideration of API design.

**Reply by MejanH** [reply]
Thanks 👍.
